internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/utils/ReportWriter.java

```diff
@@ -0,0 +1,115 @@
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class which is responsible to Generate Reports. It writes the result generated by the particular scanner tool.
+ * The content of the file will be prettify json format.
+ */
+public class ReportWriter {
+    private static final Logger log = Logger.getLogger(ReportWriter.class);
+
+    private ReportWriter() {
+    }
+
+    /**
+     * Call writer to generate report
+     *
+     * @param productResponseMapper Mapper which holds value for product and response.
+     * @param targetDir             Root Directory where the report is generated.
+     * @return Mapper which holds product and report file path key value pair.
+     * @throws FileHandlerException Exception occurred while generating report.
+     */
+    public static HashMap<String, String> callWriter(HashMap<String, String> productResponseMapper, File targetDir)
+            throws FileHandlerException {
+        String reportDesPath = targetDir.getAbsolutePath() + File.separator + java.time.LocalDate.now().toString();
+        File reportDestDir = new File(reportDesPath);
+        createDirectory(reportDestDir);
+        HashMap<String, String> reportFileMapper = new HashMap<>();
+        for (Map.Entry<String, String> entry : productResponseMapper.entrySet()) {
+            String path = writeFile(entry.getKey(), entry.getValue(), reportDestDir);
+            reportFileMapper.put(entry.getKey(), path);
+        }
+        return reportFileMapper;
+    }
+
+    /**
+     * Write resposne in file.
+     *
+     * @param name      Name of the product.
+     * @param response  Scan response.
+     * @param targetDir Directory where the report placed.
+     * @return file path
+     * @throws FileHandlerException File handler Exception.
+     */
+    private static String writeFile(String name, String response, File targetDir) throws FileHandlerException {
+        PrintWriter writer;
+        String currentReportFilePath;
+        String indented;
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            Object json = mapper.readValue(response, Object.class);
+            indented = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(json);
+            currentReportFilePath = targetDir + "/" + name + "-" +
+                    java.time.LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")) + ".json";
+        } catch (IOException e) {
+            throw new FileHandlerException("Failed to generate report for " + name + "." +
+                    " Error occurred while formatting scan results (JSON format).", e);
+        }
+        try {
+            writer = new PrintWriter(currentReportFilePath, "UTF-8");
+            writer.write(indented);
+            writer.close();
+        } catch (IOException e) {
+            throw new FileHandlerException("Failed to generate report for " + name + " " + e);
+        }
```

Comment:
The above two try blocks don't seem to give much different information upon receiving an exception. Specific messages should be used when throwing exceptions. If the same message is applicable to both, then it'd better to have the code of them inside a single try block.